momento-sdk/src/main/java/momento/sdk/SubscriptionWrapper.java

```diff
@@ -70,6 +84,16 @@ public void onError(Throwable t) {
               firstMessage = false;
               future.completeExceptionally(t);
 
+              // Retry logic for UNAVAILABLE errors
+              if (t instanceof io.grpc.StatusRuntimeException) {
```

Comment:
let's make sure we have a `logger.debug` statement in each of the branches of all of the conditions here, may be important for debugging.